Sources/TuistSupport/System/System.swift

```diff
@@ -478,6 +491,89 @@ public final class System: Systeming {
         .subscribeOn(ConcurrentDispatchQueueScheduler(queue: DispatchQueue.global()))
     }
 
+    public func observable(_ arguments: [String], pipedToArguments: [String]) -> Observable<SystemEvent<Data>> {
+        observable(arguments, environment: env, pipeTo: pipedToArguments)
+    }
+
+    public func observable(_ arguments: [String],
```

Comment:
maybe this function could be split into smaller chunks? This is little hard to digest 😐

Reply:
Completely agree! I had gone down that route with some additional methods to help readability, but opted for code organization within the method because it caused it to reference `self` in the observable that was returned. 

I'm good with breaking up with method if we don't mind having `[weak self]` in the closure that is returned? What're your thoughts?

Reply:
have you considered making those functions `static` with some additional parameters? That way we would not have to weakify self while also having a better organization

Reply:
Ooh that can definitely be done 👍🏼 Good call!